### DISCUSSION TITLE
Allow searching for multiple tags or omitting tags

### DISCUSSION THREAD

**Discussion by jacobgkau** [Kovah/LinkAce#545] [Ideas & Feature Requests]
One of the reasons tags are supposed to be more useful than folders is that they can be combined. For example, I could have three links:

- Link 1: tag1, tag2
- Link 2: tag1, tag3
- Link 3: tag2, tag3

With this collection, I should be able to search "tag1 tag3" to have only Link 2 return. Currently, if I specify two tags, I get results that only contain one of the tags.

Similarly, I should be able to search e.g. "tag2 -tag1" and have only Link 3 return.

**Comment by DrSwad** [comment]
This would be extremely helpful and kind of necessary for tags to be useful. I was expecting the tag-searching functionality to behave this way and organized my bookmarks accordingly, but just now disappointingly found out that it doesn't.

**Comment by aagd** [comment]
I also think this would be **THE** most useful improvement for search. The search query should allow `tagX AND tagY`. Now it's `tagX OR tagY` which is does not help in any way.
